packages/framework/build.xml

```diff
@@ -708,291 +885,476 @@
+                <exec executable="${path.php.executable}" passthru="true" checkreturn="true">
+                    <arg value="${path.bin-console}"/>
+                    <arg value="--env=test"/>
+                    <arg value="shopsys:domains-urls:replace"/>
+                </exec>
+            </then>
+        </if>
     </target>
 
-    <target name="test-plugin-demo-data-load" description="Loads data fixtures of all registered plugins into test DB." hidden="true">
-        <exec executable="${path.php.executable}" passthru="true" checkreturn="true">
-            <arg value="${path.bin-console}"/>
-            <arg value="--env=test"/>
-            <arg value="shopsys:plugin-data-fixtures:load"/>
-        </exec>
+    <target name="test-friendly-urls-generate" depends="production-protection" description="Generates friendly urls for supported entities when missing in test DB." hidden="true">
+        <if>
+            <equals arg1="${production.confirm.action}" arg2="y"/>
+            <then>
+                <exec executable="${path.php.executable}" passthru="true" checkreturn="true">
+                    <arg value="${path.bin-console}"/>
+                    <arg value="--env=test"/>
+                    <arg value="shopsys:generate:friendly-url"/>
+                </exec>
+            </then>
+        </if>
     </target>
 
-    <target name="test-product-search-create-structure" description="Creates structure for searching via elasticsearch for test environment." hidden="true">
-        <exec executable="${path.php.executable}" passthru="true" checkreturn="true">
-            <arg value="${path.bin-console}"/>
-            <arg value="--env=test"/>
-            <arg value="shopsys:product-search:create-structure"/>
-        </exec>
+    <target name="test-plugin-demo-data-load" depends="production-protection" description="Loads data fixtures of all registered plugins into test DB." hidden="true">
+        <if>
+            <equals arg1="${production.confirm.action}" arg2="y"/>
+            <then>
+                <exec executable="${path.php.executable}" passthru="true" checkreturn="true">
+                    <arg value="${path.bin-console}"/>
+                    <arg value="--env=test"/>
+                    <arg value="shopsys:plugin-data-fixtures:load"/>
+                </exec>
+            </then>
+        </if>
     </target>
 
-    <target name="test-product-search-delete-structure" description="Deletes structure for searching via elasticsearch for test environment." hidden="true">
-        <exec executable="${path.php.executable}" passthru="true" checkreturn="true">
-            <arg value="${path.bin-console}"/>
-            <arg value="--env=test"/>
-            <arg value="shopsys:product-search:delete-structure"/>
-        </exec>
+    <target name="test-product-search-create-structure" depends="production-protection" description="Creates structure for searching via elasticsearch for test environment." hidden="true">
```

Comment:
I have to ask once more - why this command is protected, and `product-search-delete-structure` is not? I know you need to delete structure on production, however, such an action should be protected IMHO.